substrate/frame/assets/src/lib.rs

```diff
@@ -1798,6 +1848,92 @@ pub mod pallet {
 			)?;
 			Ok(())
 		}
+
+		/// Mint a distribution of assets of a particular class.
+		///
+		/// The origin must be Signed and the sender must be the Issuer of the asset `id`.
+		///
+		/// - `id`: The identifier of the asset to have some amount minted.
+		/// - `merkle_root`: The merkle root of a compact base-16 merkle trie used to authorize
+		///   minting.
+		///
+		/// Emits `DistributionIssued` event when successful.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(33)]
+		pub fn mint_distribution(
+			origin: OriginFor<T>,
+			id: T::AssetIdParameter,
+			merkle_root: DistributionHashOf<T, I>,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			let id: T::AssetId = id.into();
+			Self::do_mint_distribution(id, merkle_root, Some(origin))?;
```

Comment:
Thinking out loud: Should we not take some kind of deposit here (charging for the storage), that is only refunded when state (`MerklizedDistributionTracker`) is cleaned up? What stops some one to create large number of distributions?